common/compat.c

```diff
@@ -169,6 +169,14 @@ getprogname (void)
 #include <fcntl.h>
 #include <unistd.h>
 
+#ifdef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+static pthread_mutex_t mutex_initializer = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+void
+p11_mutex_init (p11_mutex_t *mutex)
+{
+	memcpy (mutex, &mutex_initializer, sizeof (pthread_mutex_t));
```

Comment:
I think this is technically undefined behavior. Copying of mutex objects is not supported.

Reply:
Thank you.  The initializer is only applicable to statically allocated mutexes, right?  In p11-kit there are a few places where dynamically allocated mutexes are used; how could we initialize them?

Reply:
After going through the uses of mutexes, I realized that only `p11_library_mutex` (and possibly `p11_virtual_mutex`) needs to be recursive (see 563606efe17cbf3b84679f5e54f60b8d68ba9015). I have changed all other initialization to use the default attribute.